Explorer/Assets/Scripts/ECS/Profiling/ProfilingProvider.cs

```diff
@@ -0,0 +1,64 @@
+using Cysharp.Threading.Tasks;
+using Unity.Profiling;
+
+namespace ECS.Profiling
+{
+    /// <summary>
+    /// Profiling provider to provide in game metrics. Profiler recorder returns values in NS, so to stay consistent with it,
+    /// our most used metric is going to be NS
+    /// </summary>
+    public class ProfilingProvider  : IProfilingProvider
+    {
+        private ProfilerRecorder mainThreadTimeRecorder;
+        private const int HICCUP_THRESHOLD_IN_NS = 50_000_000;
+        private const int HICCUP_BUFFER_SIZE = 1_000;
+        private LinealBufferHiccupCounter hiccupBufferCounter;
+
+        public ProfilingProvider()
+        {
+            mainThreadTimeRecorder = ProfilerRecorder.StartNew(ProfilerCategory.Internal, "Main Thread", 15);
+            hiccupBufferCounter = new LinealBufferHiccupCounter(HICCUP_BUFFER_SIZE, HICCUP_THRESHOLD_IN_NS);
+            CountHiccup().Forget();
+        }
+
+        public long GetCurrentFrameTimeValueInNS() =>
+            mainThreadTimeRecorder.CurrentValue;
+
+        public double GetAverageFrameTimeValueInNS() =>
+            GetRecorderFPSAverage(mainThreadTimeRecorder);
+
+
+        public int GetHiccupCountInBuffer() =>
+            hiccupBufferCounter.HiccupsCountInBuffer;
+
+        private async UniTaskVoid CountHiccup()
```

Comment:
It should be handled in a system. Only systems are authorized to have loops like this